src/Yoti/Http/AbstractRequestHandler.php

```diff
@@ -2,158 +2,117 @@
-     *
-     * @throws RequestException
+     * @param string $sdkIdentifier
      */
     public function __construct($apiUrl, $pem, $sdkId = null, $sdkIdentifier = null)
     {
+        $this->apiUrl = $apiUrl;
         $this->pemFile = PemFile::fromString($pem);
-        $this->sdkId = $sdkId;
-        $this->apiUrl = rtrim($apiUrl, '/');
 
+        if (isset($sdkId)) {
+            $this->sdkId = $sdkId;
+        }
         if (isset($sdkIdentifier)) {
-            $this->setSdkIdentifier($sdkIdentifier);
-        } else {
-            $this->sdkIdentifier = self::YOTI_SDK_IDENTIFIER;
+            $this->sdkIdentifier = $sdkIdentifier;
         }
     }
 
     /**
      * @param string $endpoint
      * @param string $httpMethod
      * @param Payload|NULL $payload
-     * @param array queryParams
      *
      * @return array
      *
      * @throws RequestException
      */
-    public function sendRequest(
-        $endpoint,
-        $httpMethod,
-        Payload $payload = null,
-        array $queryParams = []
-    ) {
-        self::validateHttpMethod($httpMethod);
-
-        // Ensure endpoint always has a single leading slash.
-        $endpoint = '/' . ltrim($endpoint, '/');
-
-        $signedDataArr = RequestSigner::signRequest($this, $endpoint, $httpMethod, $payload, $queryParams);
-        $requestHeaders = $this->generateRequestHeaders($signedDataArr[RequestSigner::SIGNED_MESSAGE_KEY]);
-        $requestUrl = $this->apiUrl . $signedDataArr[RequestSigner::END_POINT_PATH_KEY];
+    public function sendRequest($endpoint, $httpMethod, Payload $payload = null)
+    {
+        $requestBuilder = (new RequestBuilder())
+          ->withBaseUrl($this->apiUrl)
+          ->withPemString((string) $this->pemFile)
+          ->withEndpoint($endpoint)
+          ->withMethod($httpMethod)
+          ->withSdkIdentifier($this->sdkIdentifier)
+          ->withQueryParam('appId', $this->sdkId);
+
+        if (isset($payload)) {
+            $requestBuilder->withPayload($payload);
+        }
 
-        return $this->executeRequest($requestHeaders, $requestUrl, $httpMethod, $payload);
-    }
+        $request = $requestBuilder->build();
 
-    /**
-     * Performs GET request.
-     *
-     * @param string $endpoint
-     * @param array queryParams
-     *
-     * @return array
-     *
-     * @throws RequestException
-     */
-    public function get($endpoint, array $queryParams = [])
-    {
-        return $this->sendRequest($endpoint, self::METHOD_GET, null, $queryParams);
-    }
+        $headers = [];
+        foreach ($request->getHeaders() as $name => $value) {
+            $headers[] = "{$name}: {$value}";
+        }
 
-    /**
-     * Performs POST request.
-     *
-     * @param string $endpoint
-     * @param Payload|NULL $payload
-     * @param array queryParams
-     *
-     * @return array
-     *
-     * @throws RequestException
-     */
-    public function post($endpoint, Payload $payload = null, array $queryParams = [])
```

Comment:
When was this introduced? Is this a breaking change or was this not released?

Reply:
This is in development but not released (was in initial signed request implementation)

Reply:
Same applies to `::get()` that's been removed in this PR